QKSMS/src/noAnalytics/java/com/moez/QKSMS/common/AnalyticsManager.java

```diff
@@ -0,0 +1,31 @@
+package com.moez.QKSMS.common;
+
+import android.content.Context;
+
+public class AnalyticsManager implements AnalyticsManagerBase {
+    private AnalyticsManager() {
+    }
+
+    private static class AnalyticsManagerHolder {
+        private final static AnalyticsManager INSTANCE = new AnalyticsManager();
+    }
+
+    public static AnalyticsManager getInstance() {
+        return AnalyticsManagerHolder.INSTANCE;
+    }
```

Comment:
Let's make `AnalyticsManagerBase` an abstract class rather than an interface, that way lines 9-15 can be moved there and don't need to be duplicated between the two `AnalyticsManager` classes